patches/api/0416-Add-whitelist-events.patch

```diff
@@ -0,0 +1,124 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: SageSphinx63920 <sage@sagesphinx63920.dev>
+Date: Sun, 14 May 2023 12:56:15 +0200
+Subject: [PATCH] Add whitelist events
+
+
+diff --git a/src/main/java/io/papermc/paper/event/player/PlayerWhitelistAddEvent.java b/src/main/java/io/papermc/paper/event/player/PlayerWhitelistAddEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..32cc41f46fdd9279750894321d0f341c3f90549e
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/event/player/PlayerWhitelistAddEvent.java
+@@ -0,0 +1,53 @@
++package io.papermc.paper.event.player;
++
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.Event;
++import org.bukkit.event.HandlerList;
++import org.jetbrains.annotations.NotNull;
++
++import java.util.UUID;
++
++/**
++ * Called when a player is being added to the whitelist
++ */
++public class PlayerWhitelistAddEvent extends Event implements Cancellable {
++
++    private static final HandlerList handlers = new HandlerList();
```

Comment:
Static final fields should be uppercase, i.e. HANDLER_LIST. Same for the other event class.